### PR TITLE
Pilot 1938: remove the redis connection

### PR DESCRIPTION
**Pull request by colorzzr**
## Summary

in order to adapt the sse, the download service will remove all direct connection with redis. Instead, it will call dataops `/v1/task` api to create/update/get job status

## JIRA Issues

PILOT-1939

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: app/resources/helpers.py [hunk 1]
```diff
@@ -124,29 +122,44 @@ async def set_status(
...
     }
-    my_value = json.dumps(record)
-    await srv_redis.set_by_key(my_key, my_value)
-    return record
+
+    # here the zipping status is the initial one. So if it occurs we
+    # create a new job otherwise update it
+    if target_status == EDataDownloadStatus.ZIPPING:
+        method = 'POST'
+        payload.update({'target_status': str(target_status)})
```
Comment (QXgu) [inline app/resources/helpers.py:148]: After taking look at the data ops /tasks API, the target_status is never being used to create the redis record. Please correct me if I am wrong

File: app/resources/helpers.py [hunk 2]
```diff
@@ -124,29 +122,44 @@ async def set_status(
...
         'job_id': job_id,
+        'task_id': job_id,
         'source': source,
         'action': action,
-        'status': str(target_status),
-        'project_code': project_code,
+        'code': project_code,
         'operator': operator,
         'payload': payload,
         'update_timestamp': str(round(time.time())),
```
Comment (QXgu) [inline app/resources/helpers.py:141]: After looking at the payload for both put and post /tasks API, the 'update_timestamp' is not the parameter required to calling the API

Comment (colorzzr) [reply app/resources/helpers.py:141]: will remove it

File: app/resources/helpers.py (hunk 1 above)
Comment (colorzzr) [reply app/resources/helpers.py:148]: yes `post` api will directly set `INIT` status. but payload still force to have the `target_status`

**Review by QXgu — APPROVED** [review]

**Pull request merged**